addons/craft/craft.lua

```diff
@@ -0,0 +1,738 @@
+--[[
+Craft v1.03312017
+
+Copyright © 2017 Mojo
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+
+* Redistributions of source code must retain the above copyright
+notice, this list of conditions and the following disclaimer.
+* Redistributions in binary form must reproduce the above copyright
+notice, this list of conditions and the following disclaimer in the
+documentation and/or other materials provided with the distribution.
+* Neither the name of craft nor the names of its contributors may be
+used to endorse or promote products derived from this software without
+specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL Mojo BE LIABLE FOR ANY
+DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+]]
+
+_addon.name     = 'craft'
+_addon.author   = 'Mojo'
+_addon.version  = '1.03312017'
+_addon.commands = {'craft'}
+
+require('chat')
+require('lists')
+require('coroutine')
+require('queues')
+require('logger')
+require('tables')
+require('sets')
+require('strings')
+
+local packets = require('packets')
+local res = require('resources')
+local recipes = require('recipes')
+local queue = Q{}
+local handlers = {}
+local delay = 24
+local synth = 0
+local skip_delay = false
+local busy = false
+local paused = false
+local display = false
+local jiggle = false
+local support = false
+local zone = nil
+
+local conditions = {
+    move = false,
+    sort = false,
+    crystal = false,
+    support = false,
+}
+
+local food = false
+local supported = false
+local appropriated = {}
+local inventory = {}
+
+local bags = {
+    safe = 1,
+    storage = 2,
+    locker = 4,
+    satchel = 5,
+    sack = 6,
+    case = 7,
+    safe2 = 9
+}
```

Comment:
Wardrobes/temporary/inventory need to be excluded from the list.  I added something that filters the resources now.